### DISCUSSION TITLE
How to implement a vertical navigation block with submenu

### DISCUSSION THREAD

**Discussion by unscripted** [WordPress/gutenberg#59619] [Q&A]
I'm struggling to create a sidebar navigation using the navigation block. I've encountered this issue a few times now and due to the challenges I faced and the timeline for the project, each time I've reverted to using a list block. This is less than ideal and I would prefer to use the navigation block.

## Steps to reproduce

1. Add a navigation block.
2. Add four menu items.
3. Add two submenu items to one of the four top-level items.
4. Change the navigation block's orientation to vertical in the settings toolbar.

## Expectation

- I would expect all submenu items to show when in vertical orientation.
- If "Open on click" is enabled, it should function like it does in the mobile nav.

Does anyone have guidance on how to successfully use the navigation block in a vertical orientation with submenus?

**Comment by ndiego** [comment]
> I would expect all submenu items to show when in vertical orientation.

Yeah, this is interesting. I guess I would not expect this by default. The Navigation should just appear vertically with the same functionality as it does horizontally. That said, I can definitely see how the layout you are looking for would be very useful in certain situations, especially sidebars.

You could probably get pretty close to what you are looking for with some custom CSS that prevents the submenu from being hidden. This could then be applied as a block style. Additionally, having a toggle or something that disables the submenu container when the Navigation block is vertically positioned could be interesting 🤔 cc @getdave

**Reply by getdave** [reply]
> I would expect all submenu items to show when in vertical orientation.

If I read this correctly, this sounds like a requirement that is specific for your use case. That's fine but we can't generalise the block in this way. Therefore it would need to be a toggle `Show submenus` or something like that. PRs welcome otherwise creating an Issue with a proposal would be fantastic 🙇
